### PR TITLE
43-convertcsvtometadataobjects-rework

### LINKED ISSUE #43: convertCSVToMetadataObjects rework

### PR DESCRIPTION
**Pull request by michalrozekariane**
Fixes #43

### CODE REVIEW

**Review by BartoszSolkaArianelabs — APPROVED** [review]

File: src/helpers/save-csv-rows-as-json-files.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { MetadataObject } from '../types/csv';
+import fs from 'fs';
+
+export const saveCSVRowsAsJsonFiles = (metadataFromCSV: MetadataObject[], folderPath: string): void => {
```
Comment (se7enarianelabs) [inline src/helpers/save-csv-rows-as-json-files.ts:4]: Rename to `saveMetadataObjectsAsJsonFiles`

File: src/csv-file-reader.ts [hunk 2]
```diff
@@ -40,10 +41,6 @@ export class CSVReaderError extends Error {
 }
 
 export class CSVFileReader {
```
[the remark this answers is not in the document]
Comment (michalrozekariane) [reply src/csv-file-reader.ts:43]: done

File: src/helpers/save-csv-rows-as-json-files.ts (hunk 1 above)
Comment (michalrozekariane) [reply src/helpers/save-csv-rows-as-json-files.ts:4]: done

File: src/services/json-metadata-from-csv-converter.ts [hunk 3]
```diff
@@ -66,11 +52,11 @@ export class JsonMetadataFromCSVConverter {
...
     csvParsedRows: CSVRow[];
-    csvFilePath: string;
+    csvFilePath?: string;
     headerAttributes: string;
     headerProperties: string;
   }): MetadataObject[] {
-    if (csvParsedRows.length <= CSVFileReader.AMOUNT_OF_HEADERS - OMITTED_HEADER_COUNT) {
+    if (csvParsedRows.length <= AMOUNT_OF_HEADERS - OMITTED_HEADER_COUNT) {
       throw new Error(dictionary.validation.csvFileIsEmpty(csvFilePath));
     }
```
[the remark this answers is not in the document]
Comment (michalrozekariane) [reply src/services/json-metadata-from-csv-converter.ts:61]: done

**Review by se7enarianelabs — APPROVED** [review]

**Pull request merged**